composer.json

```
@@ -3,15 +3,21 @@
     "description": "Contexts that we use with Behat 3.x tests on Drupal sites.",
     "type": "library",
     "license": "GPL-2.0+",
-    "require": {
-        "nuvoleweb/drupal-behat": "^1.2"
-    },
     "authors": [
         {
             "name": "Metadrop",
             "email": "hi@metadrop.net"
         }
     ],
+    "repositories": [
+      {
+        "type": "vcs",
+        "url": "https://github.com/Metadrop/drupal-behat.git"
+      }
+    ],
+    "require": {
+      "nuvoleweb/drupal-behat": "^1.2"
```

By a slack conversation it ha been told to me that These changes are supposedly compatible with drupal 9 and 10:
- In drupal 10 it would require 1.3.0
- In Drupal 9 it would require 1.2.0

But:
-  The version it is only requiring 1.2, it should require 1.2 or 1.3.
- Nobody has confirmed a composer requires work in a Drupal 9 project and tests are run successfully. A minimal manual test must be done.

```suggestion
      "nuvoleweb/drupal-behat": "^1.2 || ^1.3"
```

---

@mistermoper 
Write 
```
"nuvoleweb/drupal-behat": "^1.2 || ^1.3"
```
is the same as that 
```
"nuvoleweb/drupal-behat": "^1.2"
```

https://semver.madewithlove.com/?package=madewithlove%2Fhtaccess-cli&constraint=^1.2&stability=stable

https://getcomposer.org/doc/articles/versions.md#caret-version-range-

> The ^ operator behaves very similarly, but it sticks closer to semantic versioning, and will always allow non-breaking updates. For example ^1.2.3 is equivalent to >=1.2.3 <2.0.0 as none of the releases until 2.0 should break backwards compatibility. For pre-1.0 versions it also acts with safety in mind and treats ^0.3 as >=0.3.0 <0.4.0 and ^0.0.3 as >=0.0.3 <0.0.4.
> 
> This is the recommended operator for maximum interoperability when writing library code.

So in the MR both versions are available.

---

Nice now I understand, I forgot about that, thanks for the explanation.